File: 物件導向程式語言 C# 7答案檔/310.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CSA03
{
    class CSA03
    {
        static void Main(string[] args)
        {
            //TODO
            int num, result;

            try
            {
                num = int.Parse(Console.ReadLine());
                if (num == 0)
                {
                    throw new ArgumentException("num could not be zero");
                }

                num = Math.Abs(num);
                result = compute(num);
                Console.Write(result);
            }
            catch
            {
                Console.Write("error");
            }
            Console.ReadKey();
        }
        // ==================================================
        // Please write your code in the specified Method
        // Do NOT change the method name, type of parameter
        // ==================================================
        static int compute(int n)
        {
            //TODO
            if (n == 1)
            {
                return 2;
            }
            else
            {
                return compute(n - 1) + (3 * n);
            }
        }

    }
}

File: 程式語言 V2答案檔/C/310.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int compute(int n);

int main()
{
	int n;
	scanf("%d", &n);
	if (n > 0 && n < 1000)
	{
		printf("%d\n", compute(n));
	}
	else
	{
		printf("error\n");
	}
}

// ===================================================
// Please write your code in the specified function
// Do NOT change the function name, type of parameter
// ===================================================
int compute(int n)
{
	int i;
	int x = 0;
	for (i = 1; i < n; i++)
	{
		int count = 1;
		int num = 0;
		int tmp = i;
		while (tmp > 9)
		{
			count++;
			tmp /= 10;
		}
		tmp = i;
		while (tmp)
		{
			num += (int)(pow(tmp % 10, count) + 0.5);
			tmp /= 10;
		}
		if (num == i)
		{
			printf("%d\n", i);
			x += i;
		}
	}
	return x;
}
